ml-algorithms/src/test/java/org/opensearch/ml/engine/MLEngineTest.java

```diff
@@ -142,6 +160,33 @@ public void train_NullInput() {
         }
     }
 
+    @Test
+    public void train_NullTrainable() {
+        exceptionRule.expect(IllegalArgumentException.class);
+        MLInput mlInput = Mockito.mock(MLInput.class);
+        when(mlInput.getAlgorithm()).thenReturn(FunctionName.LINEAR_REGRESSION);
+        when(MLEngineClassLoader.initInstance(mlInput.getAlgorithm(), mlInput.getParameters(), MLAlgoParams.class)).thenReturn(null);
+        mlEngine.train(mlInput);
+    }
+
+    @Test
+    public void predict_NullTrainAndPredictable() {
+        exceptionRule.expect(IllegalArgumentException.class);
+        MLInput mlInput = Mockito.mock(MLInput.class);
+        MLModel mlModel = Mockito.mock(MLModel.class);
+        when(mlInput.getAlgorithm()).thenReturn(FunctionName.LINEAR_REGRESSION);
+        when(MLEngineClassLoader.initInstance(mlInput.getAlgorithm(), mlInput.getParameters(), MLAlgoParams.class)).thenReturn(null);
+        mlEngine.predict(mlInput, mlModel);
+    }
+
+    @Test
+    public void trainAndPredict_NullTrainable() {
+        exceptionRule.expect(IllegalArgumentException.class);
+        MLInput mlInput = Mockito.mock(MLInput.class);
+        when(mlInput.getAlgorithm()).thenReturn(FunctionName.LINEAR_REGRESSION);
+        when(MLEngineClassLoader.initInstance(mlInput.getAlgorithm(), mlInput.getParameters(), MLAlgoParams.class)).thenReturn(null);
+        mlEngine.trainAndPredict(mlInput);
+    }
     //TODO: fix mockito error
     @Ignore
     @Test
```

Comment:
@dhrubo-os we should fix this while we're at it?

Reply:
Good idea. @divitr if you think it's easy enough to fix in this PR, please go forward. Otherwise we can create another issue to take out the @ignore and make this test working.

Reply:
@dhrubo-os I don't think this issue is that easy to fix in this PR, I think we should open a new issue. I can open up a new issue if you'd like.

Reply:
Sure, I'll create an issue later.